src/dmstor.cpp

```diff
@@ -27,19 +27,19 @@ dmstor(const char *is, char **rs) {
 
 	double
 dmstor_ctx(PJ_CONTEXT *ctx, const char *is, char **rs) {
-	int n, nl;
+	int n, nl, adv;
 	char *s, work[MAX_WORK];
         const char* p;
 	double v, tv;
 
 	if (rs)
 		*rs = (char *)is;
-	/* copy sting into work space */
+	/* copy string into work space */
 	while (isspace(*is)) ++is;
 	n = MAX_WORK;
 	s = work;
 	p = (char *)is;
-	while (isgraph(*p) && --n)
+	while ((isgraph(*p) || *p == (char) 0xc2 || *p == (char) 0xb0) && --n)
```

Comment:
A comment explaining the symbols would be useful here

Reply:
Hi @kbevers, thanks for your feedback.  I have added a comment explaining this line and pushed a new commit to my branch.

Reply:
We can test more accurately for the Degree Sign with:
```suggestion
	while ((isgraph(*p) || (p[0] == (char) 0xc2 && p[1] == (char) 0xb0)) && --n)
```

Reply:
That's not equivalent; the `p[1]` may go out bounds, but also this check will only match on the first byte, and not copy the second byte.

Reply:
> `p[1]` may go out bounds

if p[0] == 0xc2, given that this is a null terminated string, p[1] cannot be out of bounds. But indeed, there would be an issue for copying the second byte.
This method should ideally be reworked to avoid the statically allocated work[] buffer and its potential overflows to use a std::string, but that's probably beyond the scope of this PR.
So yes, the current test might be acceptable

Reply:
Yeah if we are really keen to only copy 0xc2 and 0xb0 bytes when they appear in sequence, I think we'd need to rewrite these 2 lines into something more like:

```
while (n > 0) {
    if (isgraph(*p)) {
        // copy 1 character, decrement n
    } else if (n > 1 && p[0] == 0xc2 && p[1] == 0xb0) {
        // copy 2 characters, decrement n by 2
    } else {
        // exit the loop
    }
}
```

Reply:
Visual Studio is not happy about the casts.
*reinterpret_cast<const unsigned char*>(p) == 0xc2 and similarly for 0xb0 would fix that.
Similarly at line 70

Reply:
What if we got rid of the casts altogether?

E.g.
```cpp
static const char DEG_SIGN1 = '\xc2';
static const char DEG_SIGN2 = '\xb0';

...

    if ( ... *p == DEG_SIGN1 ... )
```

g++ and my local linters seem to be perfectly fine with this. I don't have a copy of VS handy to check that.

Reply:
> seem to be perfectly fine with this.

I believe that would be a good alternative. You could use "constexpr char DEG_SIGN ..."

Reply:
Thanks, I have pushed that change to the PR branch.